### PR TITLE
wasp-agent: allow swapping only for VM workloads

### PR DESCRIPTION
**Pull request by enp0s3**
Signed-off-by: Igor Bezukh <ibezukh@redhat.com>

on init disable swapping for every cgroup slice created by systemd, and also for best effort QoS cgroup

remove annotation filter in OCI hook to cactch every pod creation. This is because guaranteed pods are created under the kube root cgroup, so we can't disable swap for them globally.

### CODE REVIEW

File: contrib/agent.sh [hunk 1]
```diff
@@ -13,6 +13,12 @@ tune_system_slice() {
   # Disable swap for system.slice
   _set 0 $FSROOT/sys/fs/cgroup/system.slice/memory.swap.max
 
+  # Disable swap for user.slice
+  _set 0 $FSROOT/sys/fs/cgroup/user.slice/memory.swap.max
+
+  # Disable swap for machine.slice
+  _set 0 $FSROOT/sys/fs/cgroup/machine.slice/memory.swap.max
+
```
Comment (fabiand) [inline contrib/agent.sh:21]: machine and user slice might not always exist.

Specifically `machine.slice`.
`user.slice` is only created on user-login iiuic.

We an do this, but then we should extend `_set` to check if the path exists.

If we wanted, then we could create systemd drop-ins instead - then they would be "boot persistent", and also applied to slices even if they are not there yet.

**Review by fabiand — COMMENTED** [review]
@enp0s3 we can do this, but for next release we might want to do more.

**Comment by fabiand** [comment]
/hold

Let's hold this for now

File: contrib/agent.sh (hunk 1 above)
Comment (enp0s3) [reply contrib/agent.sh:21]: @fabiand You are correct, we might not be in sync with the creation of these slices. 

Closing this PR in favor of drop-in creation

/close

**Pull request closed without merging**